cirq/circuits/text_diagram_drawer.py

```diff
@@ -126,6 +138,42 @@ def height(self) -> int:
             max_y = max(max_y, v.y1, v.y2)
         return 1 + max_y
 
+    def force_horizontal_padding_after(self, index, padding) -> None:
+        self.horizontal_padding[index] = padding
+
+    def force_vertical_padding_after(self, index, padding) -> None:
+        self.vertical_padding[index] = padding
+
+    def shift_horizontally(self, index, offset) -> None:
```

Comment:
Create a helper method `_transform_coordinates(func: Callable[[int, int], Tuple[int ,int]])` and implement both shift methods using the helper.